pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -228,13 +261,36 @@ func (t *TopologyCache) SetNodes(nodes []*v1.Node) {
 	}
 }
 
+// HasPopulatedHints checks whether there are populated hints for a given service in the cache.
+func (t *TopologyCache) HasPopulatedHints(serviceKey string) bool {
+	return t.hintsPopulatedByService.Has(serviceKey)
+}
+
 // getAllocations returns a set of minimum and maximum allocations per zone. If
 // it is not possible to provide allocations that are below the overload
 // threshold, a nil value will be returned.
-func (t *TopologyCache) getAllocations(numEndpoints int) map[string]Allocation {
-	if t.cpuRatiosByZone == nil || len(t.cpuRatiosByZone) < 2 || len(t.cpuRatiosByZone) > numEndpoints {
-		klog.V(2).Infof("Insufficient info to allocate endpoints (%d endpoints, %d zones)", numEndpoints, len(t.cpuRatiosByZone))
```

Comment:
I think we should keep logging, in addition to the event, no?

Reply:
I do the logging based on the event we get. 
https://github.com/LiorLieberman/kubernetes/blob/7562e87620f1bb65b92b0531ecdd7e6323461499/pkg/controller/endpointslice/topologycache/topologycache.go#L96

But if I aggregate the events based on the suggestion you suggested on the next comment, should we log all of them as well?

Reply:
ic, missed that part, thanks, you can discard this comment